sympy/series/fourier.py

```diff
@@ -90,6 +91,37 @@ def _find_x(func):
     return sympify((x, start, stop))
 
 
+def finite_check(f, x, L):
+    def check_fx(exprs, x):
+        return x not in exprs.free_symbols
+
+    def check_sincos(expr, x, L):
+        if type(expr) == sin or type(expr) == cos:
+            a = Wild('a', properties=[lambda k: k.is_Integer, lambda k: k != S.Zero, ])
```

Comment:
`a` and `b` could probably be taken out of `check_sincos` so they would be created only once.

Reply:
Ok, I can do that. They might be adding some overhead.